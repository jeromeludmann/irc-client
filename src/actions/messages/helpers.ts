import { Action } from "redux";
import { Route } from "@app/Route";

export interface MessageAction<T, M> extends Action<T> {
  payload: M;
  route: Route;
}

export type MessageActionCreator<A, P = Prefix> = (
  serverKey: string,
  prefix: P,
  params: string[],
) => A;

export type Prefix = Server | User;

export type Server = string;

export type User = {
  nick: string;
  user: string;
  host: string;
};

export const isChannel = (channel: string) => /^(&|#|\+|!)/.test(channel);

export const isPrefixServer = (prefix: Prefix) => typeof prefix === "string";

export const isPrefixUser = (prefix: Prefix) => !isPrefixServer(prefix);
