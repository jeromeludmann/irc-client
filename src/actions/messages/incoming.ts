import { RoutedAction, BufferKey, isChannel } from '@app/utils/Route'
import { User, Server, Prefix, isPrefixServer } from '@app/utils/Message'

interface ReceiveMessageAction<T = string, P = {}> extends RoutedAction<T> {
  payload: P
}

export const RECEIVE_ERROR = 'MESSAGE/RECEIVE_ERROR'

export type ReceiveErrorAction = ReceiveMessageAction<
  typeof RECEIVE_ERROR,
  {
    message: string
  }
>

export const RECEIVE_JOIN = 'MESSAGE/RECEIVE_JOIN'

export type ReceiveJoinAction = ReceiveMessageAction<
  typeof RECEIVE_JOIN,
  {
    user: User
    channel: string
  }
>

export const RECEIVE_NICK = 'MESSAGE/RECEIVE_NICK'

export type ReceiveNickAction = ReceiveMessageAction<
  typeof RECEIVE_NICK,
  {
    user: User
    nick: string
  }
>

export const RECEIVE_NOTICE_FROM_SERVER = 'MESSAGE/RECEIVE_NOTICE_FROM_SERVER'
export const RECEIVE_NOTICE_FROM_CHANNEL = 'MESSAGE/RECEIVE_NOTICE_FROM_CHANNEL'
export const RECEIVE_NOTICE_FROM_USER = 'MESSAGE/RECEIVE_NOTICE_FROM_USER'

export type ReceiveNoticeFromServerAction = ReceiveMessageAction<
  typeof RECEIVE_NOTICE_FROM_SERVER,
  {
    server: Server
    text: string
  }
>

export type ReceiveNoticeFromChannelAction = ReceiveMessageAction<
  typeof RECEIVE_NOTICE_FROM_CHANNEL,
  {
    user: User
    channel: string
    text: string
  }
>

export type ReceiveNoticeFromUserAction = ReceiveMessageAction<
  typeof RECEIVE_NOTICE_FROM_USER,
  {
    user: User
    text: string
  }
>

export type ReceiveNoticeActions =
  | ReceiveNoticeFromServerAction
  | ReceiveNoticeFromChannelAction
  | ReceiveNoticeFromUserAction

export const RECEIVE_PART = 'MESSAGE/RECEIVE_PART'

export type ReceivePartAction = ReceiveMessageAction<
  typeof RECEIVE_PART,
  {
    user: User
    channel: string
    message?: string
  }
>

export const RECEIVE_PING_FROM_SERVER = 'MESSAGE/RECEIVE_PING_FROM_SERVER'

export type ReceivePingFromServerAction = ReceiveMessageAction<
  typeof RECEIVE_PING_FROM_SERVER,
  {
    key: string
  }
>

export const RECEIVE_PONG_FROM_SERVER = 'MESSAGE/RECEIVE_PONG_FROM_SERVER'

export type ReceivePongFromServerAction = ReceiveMessageAction<
  typeof RECEIVE_PONG_FROM_SERVER,
  {
    key: string
    lag: number
  }
>

export const RECEIVE_PRIVMSG = 'MESSAGE/RECEIVE_PRIVMSG'

export type ReceivePrivmsgAction = ReceiveMessageAction<
  typeof RECEIVE_PRIVMSG,
  {
    user: User
    text: string
  }
>

export const RECEIVE_RPL_MYINFO = 'MESSAGE/RECEIVE_RPL_MYINFO'

export type ReceiveReplyMyInfoAction = ReceiveMessageAction<
  typeof RECEIVE_RPL_MYINFO,
  {
    serverName: string
    version: string
    availableUserModes: string[]
    availableChannelModes: string[]
  }
>

export const messageReceivers: {
  [command: string]: (
    serverKey: string,
    prefix: Prefix | void,
    params: string[],
  ) => ReceiveMessageAction
} = {
  ERROR: (serverKey, _: void, params): ReceiveErrorAction => ({
    type: RECEIVE_ERROR,
    payload: { message: params[0] },
    route: { serverKey, bufferKey: BufferKey.ALL },
  }),

  JOIN: (serverKey, user: User, params): ReceiveJoinAction => ({
    type: RECEIVE_JOIN,
    payload: { user, channel: params[0] },
    route: { serverKey, bufferKey: params[0] },
  }),

  NICK: (serverKey, user: User, params): ReceiveNickAction => ({
    type: RECEIVE_NICK,
    payload: { user, nick: params[0] },
    route: { serverKey, bufferKey: BufferKey.STATUS },
  }),

  NOTICE: (serverKey, prefix: Prefix, params): ReceiveNoticeActions => {
    if (isPrefixServer(prefix)) {
      return {
        type: RECEIVE_NOTICE_FROM_SERVER,
        payload: { server: prefix as Server, text: params[1] },
        route: { serverKey, bufferKey: BufferKey.STATUS },
      }
    }

    return isChannel(params[0])
      ? {
          type: RECEIVE_NOTICE_FROM_CHANNEL,
          payload: {
            user: prefix as User,
            channel: params[0],
            text: params[1],
          },
          route: { serverKey, bufferKey: params[0] },
        }
      : {
          type: RECEIVE_NOTICE_FROM_USER,
          payload: { user: prefix as User, text: params[1] },
          route: { serverKey, bufferKey: BufferKey.ACTIVE },
        }
  },

  PART: (serverKey, user: User, params): ReceivePartAction => ({
    type: RECEIVE_PART,
    payload: { user, channel: params[0], message: params[1] },
    route: { serverKey, bufferKey: params[0] },
  }),

  PING: (serverKey, _: Server, params): ReceivePingFromServerAction => ({
    type: RECEIVE_PING_FROM_SERVER,
    payload: { key: params.join(' ') },
    route: { serverKey, bufferKey: BufferKey.STATUS },
  }),

  PONG: (serverKey, _: Server, params): ReceivePongFromServerAction => ({
    type: RECEIVE_PONG_FROM_SERVER,
    payload: { key: params[1], lag: 0 },
    route: { serverKey, bufferKey: BufferKey.STATUS },
  }),

  PRIVMSG: (serverKey, user: User, params): ReceivePrivmsgAction => ({
    type: RECEIVE_PRIVMSG,
    payload: { user, text: params[1] },
    route: {
      serverKey,
      bufferKey: isChannel(params[0]) ? params[0] : user.nick,
    },
  }),

  '004': (serverKey, _: Server, params): ReceiveReplyMyInfoAction => {
    const [, serverName, version] = params
    const availableUserModes = params[3].split('')
    const availableChannelModes = params[4].split('')

    return {
      type: RECEIVE_RPL_MYINFO,
      payload: {
        serverName,
        version,
        availableUserModes,
        availableChannelModes,
      },
      route: { serverKey, bufferKey: BufferKey.STATUS },
    }
  },
}
