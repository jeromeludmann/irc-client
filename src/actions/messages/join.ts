import {
  User,
  MessageAction,
  MessageActionCreator,
} from "@app/actions/messages/raw";

interface Join {
  user: User;
  channel: string;
}

export const MESSAGE_JOIN = "MESSAGE/JOIN";

export type JoinAction = MessageAction<typeof MESSAGE_JOIN, Join>;

export const joinReceived: MessageActionCreator<JoinAction, User> = (
  user,
  params,
) => {
  return {
    type: MESSAGE_JOIN,
    payload: { user, channel: params[0] },
    route: { channel: params[0] },
  };
};
