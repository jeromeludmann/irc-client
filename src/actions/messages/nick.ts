import {
  User,
  MessageAction,
  MessageActionCreator,
} from "@app/actions/messages/helpers";
import { STATUS_WINDOW } from "@app/Route";

interface Nick {
  user: User;
  nick: string;
}

export const MESSAGE_NICK = "MESSAGE/NICK";

export type NickAction = MessageAction<typeof MESSAGE_NICK, Nick>;

export const nickReceived: MessageActionCreator<NickAction, User> = (
  serverKey,
  user,
  params,
) => ({
  type: MESSAGE_NICK,
  payload: { user, nick: params[0] },
  route: { server: serverKey, window: STATUS_WINDOW },
});
