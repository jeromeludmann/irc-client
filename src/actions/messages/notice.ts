import {
  MessageAction,
  MessageActionCreator,
  isChannel,
  isPrefixServer,
  User,
  Server,
} from "@app/actions/messages/helpers";
import { STATUS_WINDOW } from "@app/Route";

interface ServerNotice {
  server: Server;
  text: string;
}

interface ChannelNotice {
  user: User;
  channel: string;
  text: string;
}

interface UserNotice {
  user: User;
  text: string;
}

export const MESSAGE_SERVER_NOTICE = "MESSAGE/SERVER_NOTICE";
export const MESSAGE_CHANNEL_NOTICE = "MESSAGE/CHANNEL_NOTICE";
export const MESSAGE_USER_NOTICE = "MESSAGE/USER_NOTICE";

export type ServerNoticeAction = MessageAction<
  typeof MESSAGE_SERVER_NOTICE,
  ServerNotice
>;

export type ChannelNoticeAction = MessageAction<
  typeof MESSAGE_CHANNEL_NOTICE,
  ChannelNotice
>;

export type UserNoticeAction = MessageAction<
  typeof MESSAGE_USER_NOTICE,
  UserNotice
>;

export const noticeReceived: MessageActionCreator<
  ServerNoticeAction | ChannelNoticeAction | UserNoticeAction
> = (serverKey, prefix, params) => {
  if (isPrefixServer(prefix)) {
    return {
      type: MESSAGE_SERVER_NOTICE,
      payload: { server: prefix as Server, text: params[1] },
      route: { server: serverKey, window: STATUS_WINDOW },
    };
  }

  return isChannel(params[0])
    ? {
        type: MESSAGE_CHANNEL_NOTICE,
        payload: { user: prefix as User, channel: params[0], text: params[1] },
        route: { server: serverKey, window: params[0] },
      }
    : {
        type: MESSAGE_USER_NOTICE,
        payload: { user: prefix as User, text: params[1] },
        route: { server: serverKey, window: STATUS_WINDOW },
      };
};
