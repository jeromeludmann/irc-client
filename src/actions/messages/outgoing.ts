import { RoutedAction, BufferKey } from '@app/utils/Route'
import { SendRawMessageAction, SEND_RAW_MESSAGE } from '@app/actions/socket'

interface SendMessageAction<T = string, P = {}> extends RoutedAction<T> {
  payload: P
}

const toRaw = (command: string, ...params: string[]): string => {
  const last = params.length - 1

  if (params[last].indexOf(' ') > -1 || params[last].charAt(0) === ':') {
    params[last] = `:${params[last]}`
  }

  return `${command.toUpperCase()} ${params.join(' ')}`.trimRight()
}

// export const SEND_JOIN = "MESSAGE/SEND_JOIN";
// export type SendJoinAction = MessageAction<typeof SEND_JOIN>;

export const sendJoin = (
  serverKey: string,
  channel: string,
): SendRawMessageAction => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('join', channel) },
  route: { serverKey, bufferKey: BufferKey.NONE },
})

// export const SEND_NICK = "MESSAGE/SEND_NICK";
// export type SendNickAction = MessageAction<typeof SEND_NICK>;

export const sendNick = (
  serverKey: string,
  nick: string,
): SendRawMessageAction => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('nick', nick) },
  route: { serverKey, bufferKey: BufferKey.NONE },
})

// export const SEND_PART = "MESSAGE/SEND_PART";
// export type SendPartAction = MessageAction<typeof SEND_PART>;

export const sendPart = (
  serverKey: string,
  channel: string,
  text: string = '',
): SendRawMessageAction => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('part', channel, text) },
  route: { serverKey, bufferKey: BufferKey.NONE },
})

export const SEND_PING_TO_SERVER = 'MESSAGE/SEND_PING_TO_SERVER'

export type SendPingToServerAction = SendMessageAction<
  typeof SEND_PING_TO_SERVER,
  {
    key: number
  }
>

export const sendPingToServer = (
  serverKey: string,
  key = Date.now(),
): SendRawMessageAction<SendPingToServerAction> => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('ping', key.toString()) },
  route: { serverKey, bufferKey: BufferKey.NONE },
  embeddedAction: {
    type: SEND_PING_TO_SERVER,
    payload: { key },
    route: { serverKey, bufferKey: BufferKey.STATUS },
  },
})

export const SEND_PONG_TO_SERVER = 'MESSAGE/SEND_PONG_TO_SERVER'

export type SendPongToServerAction = SendMessageAction<
  typeof SEND_PONG_TO_SERVER,
  {
    key: string
  }
>

export const sendPongToServer = (
  serverKey: string,
  key: string,
): SendRawMessageAction<SendPongToServerAction> => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('pong', key) },
  route: { serverKey, bufferKey: BufferKey.NONE },
  embeddedAction: {
    type: SEND_PONG_TO_SERVER,
    payload: { key },
    route: { serverKey, bufferKey: BufferKey.STATUS },
  },
})

export const SEND_PRIVMSG = 'MESSAGE/SEND_PRIVMSG'

export type SendPrivmsgAction = SendMessageAction<
  typeof SEND_PRIVMSG,
  { text: string }
>

export const sendPrivmsg = (
  serverKey: string,
  channel: string,
  text: string,
): SendRawMessageAction<SendPrivmsgAction> => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('privmsg', channel, text) },
  route: { serverKey, bufferKey: BufferKey.NONE },
  embeddedAction: {
    type: SEND_PRIVMSG,
    payload: { text },
    route: { serverKey, bufferKey: channel },
  },
})

// export const SEND_QUIT = "MESSAGE/SEND_QUIT";
// export type SendQuitAction = MessageAction<typeof SEND_QUIT>;

export const sendQuit = (
  serverKey: string,
  text: string = '',
): SendRawMessageAction => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('quit', text) },
  route: { serverKey, bufferKey: BufferKey.NONE },
})

// export const SEND_USER = "MESSAGE/SEND_USER";
// export type SendUserAction = MessageAction<typeof SEND_USER>;

export const sendUser = (
  serverKey: string,
  username: string,
  realname: string,
): SendRawMessageAction => ({
  type: SEND_RAW_MESSAGE,
  payload: { raw: toRaw('user', username, '0', '*', realname) },
  route: { serverKey, bufferKey: BufferKey.NONE },
})
