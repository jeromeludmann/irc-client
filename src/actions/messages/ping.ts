import {
  MessageAction,
  MessageActionCreator,
  Prefix,
} from "@app/actions/messages/raw";

interface Ping {
  server: Prefix;
  key: string;
}

export const MESSAGE_PING = "MESSAGE/PING";

export type PingAction = MessageAction<typeof MESSAGE_PING, Ping>;

export const pingReceived: MessageActionCreator<PingAction> = (
  prefix,
  params,
) => {
  // TODO MESSAGE_PING_SERVER
  // TODO MESSAGE_PING_USER

  return {
    type: MESSAGE_PING,
    payload: { server: prefix, key: params.join(" ") },
    route: { channel: "status" },
  };
};
