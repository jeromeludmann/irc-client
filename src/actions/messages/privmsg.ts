import {
  User,
  MessageAction,
  MessageActionCreator,
  isChannel,
} from "@app/actions/messages/helpers";

interface ChannelPrivmsg {
  user: User;
  channel: string;
  text: string;
}

interface UserPrivmsg {
  user: User;
  text: string;
}

export const MESSAGE_CHANNEL_PRIVMSG = "MESSAGE/CHANNEL_PRIVMSG";
export const MESSAGE_USER_PRIVMSG = "MESSAGE/USER_PRIVMSG";

export type ChannelPrivmsgAction = MessageAction<
  typeof MESSAGE_CHANNEL_PRIVMSG,
  ChannelPrivmsg
>;

export type UserPrivmsgAction = MessageAction<
  typeof MESSAGE_USER_PRIVMSG,
  UserPrivmsg
>;

export const privmsgReceived: MessageActionCreator<
  ChannelPrivmsgAction | UserPrivmsgAction,
  User
> = (serverKey, user, params) => {
  const target = params[0];
  const text = params[1];

  return isChannel(target)
    ? {
        type: MESSAGE_CHANNEL_PRIVMSG,
        payload: { user, channel: target, text },
        route: { server: serverKey, window: target },
      }
    : {
        type: MESSAGE_USER_PRIVMSG,
        payload: { user, text },
        route: { server: serverKey, window: user.nick },
      };
};
