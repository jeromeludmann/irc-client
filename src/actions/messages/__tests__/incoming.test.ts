import { messageReceivers } from '@app/actions/messages/incoming'
import { User } from '@app/core/Message'

describe('incoming messages actions', () => {
  const user: User = { nick: 'nick', user: 'user', host: 'host' }

  it('should receive ERROR', () => {
    expect(
      messageReceivers.ERROR('serverKey', '', ['Error message']),
    ).toMatchSnapshot()
  })

  it('should receive JOIN', () => {
    expect(
      messageReceivers.JOIN('serverKey', user, ['#channel']),
    ).toMatchSnapshot()
  })

  it('should receive NICK', () => {
    expect(
      messageReceivers.NICK('serverKey', user, ['new_nick']),
    ).toMatchSnapshot()
  })

  it('should receive NOTICE from server', () => {
    expect(
      messageReceivers.NOTICE('serverKey', 'irc.network', [
        'irc.network',
        'hello world',
      ]),
    ).toMatchSnapshot()
  })

  it('should receive NOTICE from channel', () => {
    expect(
      messageReceivers.NOTICE('serverKey', user, ['#channel', 'hello world']),
    ).toMatchSnapshot()
  })

  it('should receive NOTICE from user', () => {
    expect(
      messageReceivers.NOTICE('serverKey', user, ['me', 'hello world']),
    ).toMatchSnapshot()
  })

  it('should receive PART', () => {
    expect(
      messageReceivers.PART('serverKey', user, ['#channel']),
    ).toMatchSnapshot()
  })

  it('should receive PART with message', () => {
    expect(
      messageReceivers.PART('serverKey', user, ['#channel', 'Goodbye!']),
    ).toMatchSnapshot()
  })

  it('should receive PONG', () => {
    expect(
      messageReceivers.PONG('serverKey', 'irc.network', ['irc.network', 'key']),
    ).toMatchSnapshot()
  })

  it('should receive PRIVMSG', () => {
    expect(
      messageReceivers.PRIVMSG(
        'serverKey',
        { nick: 'nick1', user: 'user', host: 'host' },
        ['nick2', 'hello world'],
      ),
    ).toMatchSnapshot()
  })

  it('should receive 004', () => {
    expect(
      messageReceivers['004']('serverKey', 'irc.network', [
        '?',
        'irc.network',
        'version',
        'available_user_modes',
        'available_channel_modes',
      ]),
    ).toMatchSnapshot()
  })
})
