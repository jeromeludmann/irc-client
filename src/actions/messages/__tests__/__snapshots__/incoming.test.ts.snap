// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`incoming messages actions should receive 004 1`] = `
Object {
  "payload": Object {
    "availableChannelModes": Array [
      "a",
      "v",
      "a",
      "i",
      "l",
      "a",
      "b",
      "l",
      "e",
      "_",
      "c",
      "h",
      "a",
      "n",
      "n",
      "e",
      "l",
      "_",
      "m",
      "o",
      "d",
      "e",
      "s",
    ],
    "availableUserModes": Array [
      "a",
      "v",
      "a",
      "i",
      "l",
      "a",
      "b",
      "l",
      "e",
      "_",
      "u",
      "s",
      "e",
      "r",
      "_",
      "m",
      "o",
      "d",
      "e",
      "s",
    ],
    "serverName": "irc.network",
    "version": "version",
  },
  "route": Object {
    "bufferKey": "@status",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_RPL_MYINFO",
}
`;

exports[`incoming messages actions should receive ERROR 1`] = `
Object {
  "payload": Object {
    "message": "Error message",
  },
  "route": Object {
    "bufferKey": "@all",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_ERROR",
}
`;

exports[`incoming messages actions should receive JOIN 1`] = `
Object {
  "payload": Object {
    "channel": "#channel",
    "user": Object {
      "host": "host",
      "nick": "nick",
      "user": "user",
    },
  },
  "route": Object {
    "bufferKey": "#channel",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_JOIN",
}
`;

exports[`incoming messages actions should receive NICK 1`] = `
Object {
  "payload": Object {
    "nick": "new_nick",
    "user": Object {
      "host": "host",
      "nick": "nick",
      "user": "user",
    },
  },
  "route": Object {
    "bufferKey": "@status",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_NICK",
}
`;

exports[`incoming messages actions should receive NOTICE from channel 1`] = `
Object {
  "payload": Object {
    "channel": "#channel",
    "text": "hello world",
    "user": Object {
      "host": "host",
      "nick": "nick",
      "user": "user",
    },
  },
  "route": Object {
    "bufferKey": "#channel",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_NOTICE_FROM_CHANNEL",
}
`;

exports[`incoming messages actions should receive NOTICE from server 1`] = `
Object {
  "payload": Object {
    "server": "irc.network",
    "text": "hello world",
  },
  "route": Object {
    "bufferKey": "@status",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_NOTICE_FROM_SERVER",
}
`;

exports[`incoming messages actions should receive NOTICE from user 1`] = `
Object {
  "payload": Object {
    "text": "hello world",
    "user": Object {
      "host": "host",
      "nick": "nick",
      "user": "user",
    },
  },
  "route": Object {
    "bufferKey": "@active",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_NOTICE_FROM_USER",
}
`;

exports[`incoming messages actions should receive PART 1`] = `
Object {
  "payload": Object {
    "channel": "#channel",
    "message": undefined,
    "user": Object {
      "host": "host",
      "nick": "nick",
      "user": "user",
    },
  },
  "route": Object {
    "bufferKey": "#channel",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_PART",
}
`;

exports[`incoming messages actions should receive PART with message 1`] = `
Object {
  "payload": Object {
    "channel": "#channel",
    "message": "Goodbye!",
    "user": Object {
      "host": "host",
      "nick": "nick",
      "user": "user",
    },
  },
  "route": Object {
    "bufferKey": "#channel",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_PART",
}
`;

exports[`incoming messages actions should receive PONG 1`] = `
Object {
  "payload": Object {
    "key": "key",
    "lag": -1,
  },
  "route": Object {
    "bufferKey": "@status",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_PONG_FROM_SERVER",
}
`;

exports[`incoming messages actions should receive PRIVMSG 1`] = `
Object {
  "payload": Object {
    "text": "hello world",
    "user": Object {
      "host": "host",
      "nick": "nick1",
      "user": "user",
    },
  },
  "route": Object {
    "bufferKey": "nick1",
    "serverKey": "serverKey",
  },
  "type": "MESSAGE/RECEIVE_PRIVMSG",
}
`;
